Tools/clinic/clinic.py

```diff
@@ -4368,7 +4368,9 @@ def parse_arg(self, argname: str, displayname: str, *, limited_capi: bool) -> st
                     if (ptr == NULL) {{{{
                         goto exit;
                     }}}}
-                    PyBuffer_FillInfo(&{paramname}, {argname}, (void *)ptr, len, 1, 0);
+                    if (PyBuffer_FillInfo(&{paramname}, {argname}, (void *)ptr, len, 1, PyBUF_SIMPLE) < 0) {{{{
+                        goto exit;
+                    }}}}
```

Comment:
@serhiy-storchaka, how about just generating an `assert`?

Reply:
```suggestion
                    int rc = PyBuffer_FillInfo(&{paramname}, {argname}, (void *)ptr, len, 1, 0);
                    assert(rc == 0), rc;
```

Reply:
An assert should have a comment saying that we're checking a detail of `PyBuffer_FillInfo` that could be changed. 
IMO, the `if` is better. I'd be surprised if you could measure the speed difference (especially with PGO).

Reply:
I agree.